### PR TITLE
c64_cass: 17 new dumps

### PR DESCRIPTION
**Pull request by scoomby**
New working software list additions
---------------------------------------
10 Computer Hits 2 (Beau-Jolly, alt) [C64 Ultimate Tape Archive V2.0]
10 Computer Hits 3 (Beau-Jolly) [C64 Ultimate Tape Archive V2.0]
10 Computer Hits 4 (Beau-Jolly) [C64 Ultimate Tape Archive V2.0]
10th Frame (U.S. Gold) [C64 Ultimate Tape Archive V2.0]
1942 (Elite Systems) [C64 Ultimate Tape Archive V2.0]
1942 (Elite Systems, alt) [C64 Ultimate Tape Archive V2.0]
1943: The Battle of Midway (Go!) [C64 Ultimate Tape Archive V2.0]
3D Pool (Firebird) [C64 Ultimate Tape Archive V2.0]
3D Pool (Kixx) [C64 Ultimate Tape Archive V2.0]
4 Most Horror (Alternative Software) [C64 Ultimate Tape Archive V2.0]
4 Soccer Simulators (Codemasters) [C64 Ultimate Tape Archive V2.0]
4th & Inches (U.S. Gold) [C64 Ultimate Tape Archive V2.0]
5th Gear (Prism Leisure) [C64 Ultimate Tape Archive V2.0]
6 Computer Hits (Beau-Jolly) [C64 Ultimate Tape Archive V2.0]
6 Computer Hits (Beau-Jolly, alt) [C64 Ultimate Tape Archive V2.0]
6 Pak (Elite Systems) [C64 Ultimate Tape Archive V2.0]
720 (U.S. Gold) [C64 Ultimate Tape Archive V2.0]

This pull request is the first step to update the c64_cass software list to reflect the C64 Ultimate Tape Archive V2.0.
I have loaded the tapes into MAME and they all load OK.
Please let me know if there is anything else that needs to be done.  Thanks.

### CODE REVIEW

**Comment by 0kmg** [comment]
Hi scoomby. I believe it's best to format your commit message for software list additions following the patterns in: https://github.com/mamedev/build/blob/master/whatsnew/whatsnew_template.txt. The python commit message scraper looks for things like the "---" line. After that put one software list item per line with a source credit after each one in [] brackets. [C64 Ultimate Tape Archive V2.0] I guess? I don't know how smart the python script is for variations of this format.

Anyhoo, if there's more to know maybe someone else can chime in. Looking forward to your next PR. Later dude! :)

**Comment by scoomby** [comment]
Thanks for the comments Bob.  I have added square brackets to cover the source credit.  I have checked recent whatsnew.txt files and typically each new software list addition for a particular device are separated by a comma rather than one software item per line.  This is probably due to the fact there can be large number of new items added on each release.  It would be good to get this confirmed though.

I have just noticed that the original "10 Computer Hits 2 (Beau-Jolly)" was incorrectly credited to the Ultimate Tape Archive V1.0.  The CRC and SHA1 do not match the V1.0 set and they have changed on the V2.0 set meaning that the 10 Computer Hits 2 should be considered an alternative set rather than a replacement.  I will look at this more closely tomorrow.

**Comment by 0kmg** [comment]
The comma separated lists in the post-processed whatsnew for each release is generated by a python script AFAICT. If you list them one-per-line the script will do that for you. Plus a commit with one software item per line is generally easier to know how many and what items are being added at a glance. Anyway, all the long-time contributors split them into separate lines so it seems the way to go even if the python script is smart enough to grab a comma separated list.

See ya next commit! ;)

**Comment by scoomby** [comment]
Hi Bob.  Thanks for information.  I have updated the pull request as advised.  I have been looking through some of your pull requests :)

I have also re-instated the original 10 Computer Hits 2 tap files and made the tap files from the C64 Ultimate Tape Archive an alternative set.

**Comment by 0kmg** [comment]
Well, I'm still learning the ropes. Look at a long-time contributor like rzero9's commits for a better idea of how to update a softwarelist.

So, I looked in your commit and have more free tips :) You should make 10hits2a, 1942a, and 6hitsa clones of their respective non-a sets, e.g.
`_<software name="1942a" cloneof="1942"`
If 3dpoolk and 5thgearp are just releases of the same software by separate publishers they can be made clones too. Don't make them clones if they are independently developed software that happen to share the same name though.

**Comment by scoomby** [comment]
Thanks Bob for the tips.  You have been a great help.  There is indeed a lot to learn!

**Comment by 0kmg** [comment]
One more: if 1943 crashes and is not playable it should be marked as not supported perhaps with an optional comment above or beside it, e.g.:
`<software name="1943" supported="no"> <!-- crashes when starting new game -->`

You should then change the commit message by adding a "New NOT_WORKING software list additions" section. Also FYI, there's also a supported="partial" tag for SL entries that are playable yet have say graphics glitches or non-working features. These are generally put in the "working" part of the commit message as there's no separate category for whatsnew info.

**Comment by scoomby** [comment]
Thanks again Bob for the comments and clarifications.  I have updated this PR as advised :)

**Comment by Tafoid** [comment]
@scoomby Unable to confirm the data to support this software list addition was sent so this can be verified.  Please use instructions @ [THIS LINK](https://wiki.mamedev.org/index.php/Submitting_Source_Code) and send links or data.

**Comment by scoomby** [comment]
Thanks Scott.  I know what to do from now on.  Regards.  Steve

**Comment by scoomby** [comment]
Whilst testing the next batch of new dumps, I realised what was causing 1943 to crash (operator error).  Further testing confirms that new dump for 1943 is working correctly in MAME

**Comment by scoomby** [comment]
Hi Bob.  Thanks for the tip.  It wasn't actually intentional to sync with the master.  I was trying anything I could to recover my work-in-progress 30 new dumps PR that I have completely lost.  My PC decided to restart itself whilst watching TV last night to install some updates which lost the online changes to the new PR.  Not a problem I thought initially as I had backed it up offline, but then I checked XML back-up containing these changes, the file was corrupted.  Arrgggh!

**Comment by Tafoid** [comment]
I've gone through this PR and it passes the -validate and xmllint scans.  However, there are a number of other issues which need addressing concerning labels and hashes as detailed below.  Please correct these and always try to double check your work before sending your Pull Requests.

-----

Both Side 1 And Side 2 have the same label.  They should be different labels, especially if sitting in the same setname, despite the different hashes.

Set:	6 Pak
Name:	6pak
ROM:	6_Pak_Side_1.tap

-----

The following entries have (alt) sets or different manufacturer releases which you have the same labels as the original (parent) set.
This is a problem when running merged sets, where all data files would be stored in the parent's setname - all of them should have unique labels.

Set:	10 Computer Hits 2 (alt)
Name:	10hits2a
ROM:	10_Computer_Hits_2_Tape_A.tap
ROM:	10_Computer_Hits_2_Tape_B.tap

Set:	3D Pool (Kixx)
Name:	3dpoolk
ROM:	3D_Pool.tap

Set:	5th Gear (Prism)
Name:	5thgearp
ROM:	5th_Gear.tap

-----

The following entries have problems where the hashes do not match the actual data provided.

- Problem:  SHA1s appear swapped between Tape 1 and 2. CRC32s for both appears correct.

10 Computer Hits 3 [folder: 10hits3]
missing set: 10 Computer Hits 3
missing rom: 10_Computer_Hits_3_Tape_1.tap [size: 1627377] [CRC32: b32d52df] [SHA1: c0da12e89bf2a78e4c42f9e4ee97776d36746ba4]
missing rom: 10_Computer_Hits_3_Tape_2.tap [size: 2323293] [CRC32: b108b920] [SHA1: 0a4e5ea9df2d4a5d2d6c6362c61f369b3a20bfc6]

Name:	10_Computer_Hits_3_Tape_2.tap
Size:	1627377
CRC32:	b32d52df
SHA1:	0a4e5ea9df2d4a5d2d6c6362c61f369b3a20bfc6

Name:	10_Computer_Hits_3_Tape_1.tap
Size:	2323293
CRC32:	b108b920
SHA1:	c0da12e89bf2a78e4c42f9e4ee97776d36746ba4

- Problem:  CRC32 is correct, SHA1 appears to match entry for "1985 - The Day After" and needs correcting.

1942 (alt) [folder: 1942a - parent: 1942]
missing set: 1942 (alt)
missing rom: 1942_a1.tap [size: 718105] [CRC32: 1a156870] [SHA1: ef8fd74086dbbfe10935f096e6839ab48dc7d5b8]

Name:	1942_a1.tap
Size:	718105
CRC32:	1a156870
SHA1:	ef99ae65f2c4b9d4d3b66cc09328b64cb7489a98

**Comment by scoomby** [comment]
Thanks Scott.  I will perform more double checking in future.  Regards.  Steve

File: hash/c64_cass.xml [hunk 1]
```diff
@@ -153,6 +337,76 @@ Some are based on the C64 Tape Archive at https://archive.org/download/c64tapes/
...
+		<part name="cass2" interface="cbm_cass">
+			<feature name="part_id" value="Side B: Jet Set Willy II / Sacred Armour of Antiriad / Fighting Warrior / Split Personalities"/>
+			<dataarea name="cass" size="2419415">
+				<rom name="6_Pak_Side_2.tap" size="2419415" crc="0ba93379" sha1="a3ae97aa38974642ee391002351882d1aeaa72c3"/>
+			</dataarea>
+		</part>
+	</software>
+
+	<software name="720">
+		<description>720</description>
```
Comment (cuavas) [inline hash/c64_cass.xml:399]: Is this actually a game called “720” or is this a variant of the skateboarding game “720°” for C64?

Comment (angelosa) [reply hash/c64_cass.xml:399]: It's the US Gold conversion of 720° most likely

Comment (scoomby) [reply hash/c64_cass.xml:399]: It is the US Gold conversion of 720 Degrees.  Description probably needs to be updated to show 720 Degrees in the same manner as the arcade version in MAME to be consistent.

**Pull request merged**